src/Http/Middleware/CspMiddleWare.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         3.6.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Http\Middleware;
+
+use Cake\Core\Configure;
+use Cake\Http\Response;
+use Cake\Http\ServerRequest;
+use InvalidArgumentException;
+use ParagonIE\CSPBuilder\CSPBuilder;
+
+/**
+ * Content Security Policy Middleware
+ */
+class CspMiddleware
+{
+
+    /**
+     * CSP Builder
+     *
+     * @var \ParagonIE\CSPBuilder\
+     */
+    protected $csp;
+
+    /**
+     * Constructor
+     *
+     * @param null|\ParagonIE\CSPBuilder\
```

Comment:
Shouldn't this include `array` too?

Reply:
Yes, I've had the idea of supporting an array after I wrote that. ;)